lib/styleToFilters.js

```diff
@@ -18,7 +21,7 @@
 function styleToFilters(style) {
   var layers = {};
   // Store layers and filters used in style
-  if (style.layers) {
+  if (style && style.layers) {
```

Comment:
What is the reason for this change @zmofei?

Reply:
Some time we may get the style from some where, it could be `null` or `underfined`. 
I know for this situation we have 2 options:
1. Through an Error to tell the user that this is illegal.
2. Do nothing.
Both of this 2 options are work I think, but the opt 1 may stop the rest of the code to run, And the opt 2 can tolerate `Non-fatal` error, it will not block the code to run and return empty `{}`, may friendly for the user and less code for use to write.

Reply:
Great, thanks for explaining. I vote for `Throw an Error to tell the user that this is illegal.`. So code like:

```js
function styleToFilters(style) {
  var layers = {};
  // Store layers and filters used in style
  if (!style || !style.layers) {
      /// throw to avoid invalid usage
  }
}
```

While returning `{}` does seem like valid behavior I think it is preferable to have a way to present an error when this condition happens. Because, I can't think of a situation where `style` would be null and we would not, as developers, want to know about it. Can you?

Reply:
👋 hey, Dane, I am trying to add the Erro, but after I added, I saw this test group in `styleToFilter.test.js#L13-L20`:

https://github.com/mapbox/vtshaver/blob/983b1e1a73429df327c7efbfd31bd49883892f22/test/styleToFilter.test.js#L13-L21

Thus, I think we should keep the original version without adding the Error. Otherwise, we need to change this test group, how do you think?

Reply:
Oh, I see. I was not aware of those tests. Okay to keep without error then.